base/bitarray.jl

```diff
@@ -811,21 +811,32 @@ function sizehint!(B::BitVector, sz::Integer)
     return B
 end
 
-resize!(B::BitVector, n::Integer) = _resize_int!(B, Int(n))
-function _resize_int!(B::BitVector, n::Int)
+resize!(B::BitVector, n::Integer; first::Bool=false) = _resize_int!(B, Int(n), first)
+function _resize_int!(B::BitVector, n::Int, first::Bool)
     n0 = length(B)
     n == n0 && return B
     n >= 0 || throw(BoundsError(B, n))
     if n < n0
-        deleteat!(B, n+1:n0)
+        r = first ? (1:n0-n) : (n+1:n0)
+        deleteat!(B, r)
         return B
     end
     Bc = B.chunks
     k0 = length(Bc)
     k1 = num_bit_chunks(n)
     if k1 > k0
-        _growend!(Bc, k1 - k0)
-        Bc[end] = UInt64(0)
+        if first
+            if (n0 & 63 == 0) && (n & 63 == 0)
```

Comment:
This optimization could be applied more often:
```suggestion
            if (n0 - n) & 63 == 0
```

Reply:
seems that it doesn't work

```julia
julia> using Random

julia> a = bitrand(10);

julia> b = copy(a);

julia> Base._growbeg!(a.chunks, 2);

julia> a.len = 74;

julia> a[end-9:end] == b
false
```

Reply:
my mistake, you are right